src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -366,6 +366,25 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
 #endif
             }
 
+            if (codeSize < _code.Length)
+            {
+                if (_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X64 ||
+                    _compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X86)
+                {
+                    // For xarch, the generated code is sometimes smaller than the memory allocated.
```

Comment:
I think this is also true for arm32: I think there are situations where we can over-allocate for arm32 if long branches can opportunistically be made shorter during emission, even if branch shortening didn't catch it. (There is no issue with misestimated instruction sizes except for choosing appropriate branch instruction size)

Reply:
Ok, I never collected numbers for arm32 so was not aware. I will update the condition.